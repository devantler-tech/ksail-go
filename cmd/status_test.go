package cmd_test

import (
	"testing"

	"github.com/devantler-tech/ksail-go/cmd"
)

func TestNewStatusCmd(t *testing.T) {
	t.Parallel()

	cmd.TestSimpleCommandCreation(t, cmd.SimpleCommandTestData{
		CommandName:   "status",
		NewCommand:    cmd.NewStatusCmd,
		ExpectedUse:   "status",
		ExpectedShort: "Show status of the Kubernetes cluster",
	})
}

func TestStatusCmd_Execute(t *testing.T) {
	t.Parallel()

	cmd.TestSimpleCommandExecution(t, cmd.SimpleCommandTestData{
		CommandName: "status",
		NewCommand:  cmd.NewStatusCmd,
	})
}

func TestStatusCmd_Help(t *testing.T) {
	t.Parallel()

	cmd.TestSimpleCommandHelp(t, cmd.SimpleCommandTestData{
		CommandName: "status",
		NewCommand:  cmd.NewStatusCmd,
	})
}
