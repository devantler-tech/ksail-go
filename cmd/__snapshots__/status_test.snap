[TestStatusCmd_Help - 1]
Show the current status of the Kubernetes cluster.

Usage:
  status [flags]

Flags:
  -c, --context string      Kubernetes context to check status for (default "kind-ksail-default")
  -h, --help                help for status
  -k, --kubeconfig string   Path to kubeconfig file (default "~/.kube/config")
  -t, --timeout duration    Timeout for status check operations (default 5m0s)

---

[TestStatusCmd_Execute - 1]
✔ Cluster status: Running (stub implementation)
► Context: kind-ksail-default
► Kubeconfig: ~/.kube/config

---